### ISSUE TITLE
Entities with computed columns do not validate when UseDataAnnotations = true

### ISSUE THREAD

**Issue by staff0rd** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#331]
For this table;

```sql
CREATE TABLE [dbo].[MyEntity]
(
	Id INT NOT NULL PRIMARY KEY IDENTITY(1,1),
	MyColumn varchar(10) NOT NULL,
	MyComputedColumn AS MyColumn
)
```

When `UseDataAnnotations = false`, then the following is generated;
```csharp
#region POCO classes

// MyEntity
[System.CodeDom.Compiler.GeneratedCode("EF.Reverse.POCO.Generator", "2.32.0.0")]
public class MyEntity
{
    public int Id { get; set; } // Id (Primary key)
    public string MyColumn { get; set; } // MyColumn (length: 10)
    public string MyComputedColumn { get; private set; } // MyComputedColumn (length: 10)
}

#endregion

#region POCO Configuration

// MyEntity
[System.CodeDom.Compiler.GeneratedCode("EF.Reverse.POCO.Generator", "2.32.0.0")]
public class MyEntityConfiguration : System.Data.Entity.ModelConfiguration.EntityTypeConfiguration<MyEntity>
{
    public MyEntityConfiguration()
        : this("dbo")
    {
    }

    public MyEntityConfiguration(string schema)
    {
        ToTable("MyEntity", schema);
        HasKey(x => x.Id);

        Property(x => x.Id).HasColumnName(@"Id").HasColumnType("int").IsRequired().HasDatabaseGeneratedOption(System.ComponentModel.DataAnnotations.Schema.DatabaseGeneratedOption.Identity);
        Property(x => x.MyColumn).HasColumnName(@"MyColumn").HasColumnType("varchar").IsRequired().IsUnicode(false).HasMaxLength(10);
        Property(x => x.MyComputedColumn).HasColumnName(@"MyComputedColumn").HasColumnType("varchar").IsRequired().IsUnicode(false).HasMaxLength(10).HasDatabaseGeneratedOption(System.ComponentModel.DataAnnotations.Schema.DatabaseGeneratedOption.Computed);
    }
}

#endregion
```

And the following test will **pass** ([Source](https://github.com/staff0rd/ef-reverse-poco-computed/tree/ca3825bf19c7259647f87c0bcebefdefcc211451));
```csharp
[TestMethod]
public void Validation_WhenEntityHasComputedColumn_ShouldValidate()
{
    var entity = new MyEntity { MyColumn = "something" };
    var context = new MyDbContext();
    context.MyEntities.Add(entity);
    context.GetValidationErrors()
        .SelectMany(p => p.ValidationErrors).Select(p => $"{p.PropertyName} - ${p.ErrorMessage})")
        .ShouldBeEmpty();
}
```
However. when setting `UseDataAnnotations = true`, the following is generated;
```csharp
#region POCO classes

// MyEntity
[System.CodeDom.Compiler.GeneratedCode("EF.Reverse.POCO.Generator", "2.32.0.0")]
public class MyEntity
{
    [Column(@"Id", Order = 1, TypeName = "int")]
    [Required]
    [Key]
    [Display(Name = "Id")]
    public int Id { get; set; } // Id (Primary key)

    [Required]
    [MaxLength(10)]
    [StringLength(10)]
    [Display(Name = "My column")]
    public string MyColumn { get; set; } // MyColumn (length: 10)

    [Required]
    [MaxLength(10)]
    [StringLength(10)]
    [Display(Name = "My computed column")]
    public string MyComputedColumn { get; private set; } // MyComputedColumn (length: 10)
}

#endregion

#region POCO Configuration

// MyEntity
[System.CodeDom.Compiler.GeneratedCode("EF.Reverse.POCO.Generator", "2.32.0.0")]
public class MyEntityConfiguration : System.Data.Entity.ModelConfiguration.EntityTypeConfiguration<MyEntity>
{
    public MyEntityConfiguration()
        : this("dbo")
    {
    }

    public MyEntityConfiguration(string schema)
    {
        ToTable("MyEntity", schema);
        Property(x => x.Id).HasDatabaseGeneratedOption(DatabaseGeneratedOption.Identity);
        Property(x => x.MyColumn).HasColumnName(@"MyColumn").HasColumnType("varchar").IsUnicode(false);
        Property(x => x.MyComputedColumn).HasColumnName(@"MyComputedColumn").HasColumnType("varchar").IsUnicode(false).HasDatabaseGeneratedOption(DatabaseGeneratedOption.Computed);
    }
}

#endregion
```
And the test will now **fail** ([Source](https://github.com/staff0rd/ef-reverse-poco-computed/tree/ca3825bf19c7259647f87c0bcebefdefcc211451)), with;
> MyComputedColumn - The My computed column field is required.

In this case perhaps the `[Required]` annotation should not be present on the computed column?

**Comment by sjh37** [comment]
An excellent bug report. Very clear, thanks.

**Comment by sjh37** [comment]
Thanks @staff0rd.
All fixed. This will be in the next release.

**Comment by sjh37** [comment]
Released in v2.34.0
